modules/apps/jaggery/mdm/src/mdm/modules/user.js

```diff
@@ -540,5 +540,42 @@ var userModule = function () {
         }
     };
 
+    /**
+     * Private method to be used by addUser() to
+     * retrieve secondary user stores.
+     * This needs Authentication since the method access admin services.
+     *
+     * @returns {string array} Array of secondary user stores.
+     */
+    publicMethods.getSecondaryUserStores = function () {
+        var ws = require('ws');
+    	var headers = [{name : "Authorization", value : "Basic YWRtaW46YWRtaW4="}];
+        var wsRequest = new ws.WSRequest();
+        var options = [];
+        options.HTTPHeaders = headers;
+        options.useSOAP = 1.2;
+        options.useWSA = 1.0;
+        options.action = "urn:getSecondaryRealmConfigurations";
+        var wsPayload = "<xsd:getSecondaryRealmConfigurations  xmlns:xsd='http://org.apache.axis2/xsd'/>";
+        var wsResponse;
+        try {
+    	    wsRequest.open(options, mdmProps.carbonServer + "services/UserStoreConfigAdminService.UserStoreCon" +
+    	        "figAdminServiceHttpsSoap12Endpoint/", false);
+    	    wsRequest.send(wsPayload);
+    	    wsResponse = wsRequest.responseE4X;
+        } catch (e) {
+    		log.error("Carbon server error while retrieving secondary user stores :" + e.toString());
+    	}
+    	var domainIDs = stringify(wsResponse.*::['return'].*::domainId.text());
+    	if (domainIDs != "\"\"") {
+    	    var characterNeedReplacing = "\"";
+            var regExpForSearch = new RegExp(characterNeedReplacing, "g");
```

Comment:
the regExp only instantiate on the page load.and this is regExp is not a huge resource hungry complicated one.As for the discussion had offline this kept unchanged.